### PR TITLE
stub files generated

### PR DESCRIPTION
**Pull request by crcrpar**
A try of https://pypi.org/project/pybind11-stubgen/ for stub files.

It'd be better to add something to call `pybind11-stubgen` in setup.py to avoid stubs being out of sync with actual bindings.

```
root@32f61dfc1c6e:/opt/pytorch/nvfuser# PYTHONPATH=. pybind11-stubgen nvfuser
pybind11_stubgen - [  ERROR] In nvfuser._C.Tensor._get_fusion_definition : Invalid expression 'nvfuser::python_frontend::FusionDefinition'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.Operators.iota : Invalid expression '<DataType.Int: 4>'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.Operators.max : Invalid expression '<DataType.Null: 13>'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.Operators.min : Invalid expression '<DataType.Null: 13>'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.Operators.normal : Invalid expression '<DataType.Float: 1>'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.Operators.prod : Invalid expression '<DataType.Null: 13>'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.Operators.sum : Invalid expression '<DataType.Null: 13>'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.Operators.uniform : Invalid expression '<DataType.Float: 1>'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.define_constant : Invalid expression 'nvfuser::StructHandle'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.define_constant : Invalid expression 'nvfuser::Pointer'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.define_constant : Invalid expression 'nvfuser::Opaque'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.define_constant : Invalid expression 'dynamic_type::DynamicType<dynamic_type::Containers<std::vector>'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.define_constant : Invalid expression 'at::Tensor'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.define_constant : Invalid expression 'std::complex<double>'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.define_constant : Can't find/import 'double'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.define_constant : Can't find/import 'long'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.define_constant : Invalid expression 'bool>'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.define_scalar : Invalid expression '<DataType.Double: 0>'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.define_scalar : Invalid expression 'nvfuser::StructHandle'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.define_scalar : Invalid expression 'nvfuser::Pointer'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.define_scalar : Invalid expression 'nvfuser::Opaque'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.define_scalar : Invalid expression 'dynamic_type::DynamicType<dynamic_type::Containers<std::vector>'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.define_scalar : Invalid expression 'at::Tensor'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.define_scalar : Invalid expression 'std::complex<double>'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.define_scalar : Can't find/import 'double'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.define_scalar : Can't find/import 'long'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.define_scalar : Invalid expression 'bool>'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.define_tensor : Invalid expression '<DataType.Float: 1>'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.define_vector : Invalid expression '<DataType.Int: 4>'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.ops. : Invalid expression 'nvfuser::python_frontend::FusionDefinition::Operators'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.ops. : Invalid expression 'nvfuser::python_frontend::FusionDefinition::Operators'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.sched. : Invalid expression 'nvfuser::python_frontend::FusionDefinition::SchedOperators'
pybind11_stubgen - [  ERROR] In nvfuser._C._FusionDefinition.sched. : Invalid expression 'nvfuser::python_frontend::FusionDefinition::SchedOperators'
pybind11_stubgen - [WARNING] Enum-like str representations were found with no matching mapping to the enum class location.
Use `--enum-class-locations` to specify full path to the following enum(s):
 - DataType
pybind11_stubgen - [WARNING] Raw C++ types/values were found in signatures extracted from docstrings.
Please check the corresponding sections of pybind11 documentation to avoid common mistakes in binding code:
 - https://pybind11.readthedocs.io/en/latest/advanced/misc.html#avoiding-cpp-types-in-docstrings
 - https://pybind11.readthedocs.io/en/latest/advanced/functions.html#default-arguments-revisited
```

### CODE REVIEW

File: .lintrunner.toml [hunk 1]
```diff
@@ -5,6 +5,7 @@ exclude_patterns = [
     '.git/**',
     'third_party/**',
     'tools/examples/**',
+    'stubs/**'
```
Comment (jjsjann123) [inline .lintrunner.toml:8]: seems like we should have excluded stub from every linter matching `**/*.py`.
A brief scan shows MYPI and MYPISTRICT not excluding those, I guess you are just turning them off on whichever is failing?

**Review by jjsjann123 — COMMENTED** [review]
I don't have a strong opinion against this, since it doesn't have real impact on functionality. Also I don't use any fancy IDE.

I do think that, instead of pushing out-dated stub files to the code base, it makes more sense to just have the generation command added to our build script and to have the pip package in requirement as well. Instead of expecting folks to manually run the update, because in that case, what's the point of having them here in the first place?

**Comment by naoyam** [comment]
Could someone explain what this PR is about for those who are not savvy with python and C++ interface?

**Comment by jjsjann123** [comment]
> Could someone explain what this PR is about for those who are not savvy with python and C++ interface?

I thought you use vs code as well :laughing: 

IIUC, this PR is just trying to add the generated stub files for all the python generated APIs we have defined in C++. So your local IDE would be able to pick up the data structures.

**Comment by crcrpar** [comment]
> I do think that, instead of pushing out-dated stub files to the code base, it makes more sense to just have the generation command added to our build script and to have the pip package in requirement as well.

it'd be too much time consuming to me.

**Pull request closed without merging**